src/spacing.ts

```diff
@@ -52,13 +52,13 @@ const SELECTOR = SELECTOR_ALLOWED
   .map((allowed) => `${allowed}:not(${SELECTOR_BLOCKED
     .flatMap((blocked) =>
       // Not include itself if it is a tag selector
-      blocked[0].match(/[a-z]/i) ? `${blocked} *` : [blocked, `${blocked} *`]
+      (blocked[0] as string).match(/[a-z]/i) ? `${blocked} *` : [blocked, `${blocked} *`],
```

Comment:
Is this required? I didn't check it locally but from my assumption it should be `string`.

Reply:
TypeScript infers that `blocked[0]` may be undefined, but I believe it is always a string. This is also to resolve the error of the `@typescript-eslint/recommended-type-checked` rule.